chainstate/src/detail/chainstateref.rs

```diff
@@ -168,18 +168,29 @@ impl<'a, S: BlockchainStorageRead, O: OrphanBlocks> ChainstateRef<'a, S, O> {
         self.db_tx.get_block_index(block_id).map_err(PropertyQueryError::from)
     }
 
-    pub fn get_gen_block_index(
-        &self,
-        block_id: &Id<GenBlock>,
+    pub fn gen_block_index_getter(
```

Comment:
Should this function be public? It feels like an implementation detail.

Reply:
Not really. There's a TODO related to this. Due to the usage in different contexts this may be used in other places, just like `block_index_ancestor_getter()`.

Reply:
Boring Saturday. I fixed them all. I'm resolving this.